tools/perplexity/perplexity.cpp

```diff
@@ -1542,12 +1542,14 @@ static void multiple_choice_score(llama_context * ctx, const common_params & par
         // the common prefix is shared among the 4 sequences to save tokens
         // we extract logits only from the last common token and from all ending tokens of each sequence
         int s0 = 0;
+        int max_seq_exceeded = 0;
         while (n_cur + (int) tasks[i1].required_tokens <= n_ctx) {
             auto& cur_task = tasks[i1];
             int n_logits = 0;
 
             int num_answers = cur_task.seq_tokens.size();
             if (s0 + num_answers > max_seq) {
+                max_seq_exceeded = s0 + num_answers;
```

Comment:
We can log the error here and straight up return. This way no need to declare `max_seq_exceeded` var

Reply:
Done.

Reply:
I guess I should check `s0 == 0` instead of `i0 == i1`.